pkg/admission/server.go

```diff
@@ -0,0 +1,142 @@
+package admission
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	admission "k8s.io/api/admission/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"net/http"
+	"path/filepath"
+	sav1 "sigs.k8s.io/service-apis/apis/v1alpha1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	tlsDir      = `/run/secrets/tls`
+	tlsCertFile = `tls.crt`
+	tlsKeyFile  = `tls.key`
+)
```

Comment:
Are there any standard tools that help simplify this part of setting up a webhook? I seem to remember kubebuilder having some relatively helpful tools here but that also added a lot of extra things I didn't need/want at the time.

Reply:
I was hoping that was the case as well. Kubebuilder does not appear to work at all within the repo when I tried to run it. I get the following 

```
kubebuilder create webhook --group networking.x-k8s.io --version v1alpha1 --kind GatewayClass --defaulting --programmatic-validation
2020/12/18 11:42:04 failed to create webhook: unable to find configuration file, project must be initialized

```